### DISCUSSION TITLE
How to quote arguments correctly in Zed Preview 0.132.0 ?

### DISCUSSION THREAD

**Discussion by failable** [zed-industries/zed#10732] [Feature Requests]
I use the below config to spawn task to open `Magit` in Emacs. But after upgrading to Zed Preview 0.132.0, it does not work.

```
  {
    "label": "Magit",
    "command": "osascript",
    "args": [
      "-e",
      "'activate application \"Emacs\"'",
      "-e",
      "'do shell script \"/usr/local/bin/emacsclient -n -u -e \\\"(magit-status)\\\"\"'"
    ],
    "use_new_terminal": false,
    "allow_concurrent_runs": false,
    "reveal": "never"
  }
```

Some errors happen like 

```
0:1: syntax error: A unknown token can’t go here. (-2740)

⏵ Task `Magit` finished with non-zero error code: 1
⏵ Command: '/usr/local/bin/bash -i -c osascript -e "'activate application \"Emacs\"'" -e "'do shell script \"/usr/local/bin/emacsclient -n -u -e \\\"(magit-status)\\\"\"'"'
```

I guess there are some argument quoting issues there. How to fix them? Thanks.

**Comment by SomeoneToIgnore** [comment]
https://github.com/zed-industries/zed/pull/10548/files#diff-463dfa15a4c73c6dfbc3d8f18897fafa8edfeef9d671d85db449db44769511e6R322
This is the culprit, and I've added that to support commands like 
```
⏵ Command: '/bin/zsh -i -c python3 -c "exec(r'''a = int(input())print(a + 3)''')"'
```
that are being generated out of templates that insert selection inside the script: https://github.com/zed-industries/zed/pull/10548/files#diff-0f9876cce13af797a382f504832e563e38123d50dedd84f3f56607a03f8ae354R189-R196

We could add a parameter to control when the arguments are escaped, but maybe @osiewicz has any good ideas how to merge the two worlds together?

**Reply by osiewicz** [reply]
I wonder if we could use heredoc instead of escaping?

**Reply by SomeoneToIgnore** [reply]
Thanks to @osiewicz , we now have a fix without any extra parameters needed: https://github.com/zed-industries/zed/pull/10776
Will pick it into the next preview patch release.
